engine/table/src/test/java/io/deephaven/engine/table/impl/QueryTableNegativeZeroTest.java

```diff
@@ -0,0 +1,459 @@
+            final Table actual = x.aggAllBy(AggSpec.unique());
+            floatToBitsEquals(actual, "X", oneKey(actual), -0.0f);
+        }
+        {
+            final Table x = newTable(floatCol("X", 0.0f, -0.0f));
+            final Table actual = x.aggAllBy(AggSpec.unique());
+            floatToBitsEquals(actual, "X", oneKey(actual), 0.0f);
+        }
+    }
+
+    @Test
+    @Ignore(value = "io.deephaven.chunk.WritableChunk.sort() is unstable wrt -0.0 & 0.0")
+    public void testDhc3768_unique_double_agg_results() {
+        {
+            final Table x = newTable(doubleCol("X", -0.0, 0.0));
+            final Table actual = x.aggAllBy(AggSpec.unique());
+            doubleToBitsEquals(actual, "X", oneKey(actual), -0.0);
+        }
+        {
+            final Table x = newTable(doubleCol("X", 0.0, -0.0));
+            final Table actual = x.aggAllBy(AggSpec.unique());
+            doubleToBitsEquals(actual, "X", oneKey(actual), 0.0);
+        }
+    }
+
+    @Test
+    public void testDhc3768_sort_float_results() {
+        {
+            final Table x = newTable(floatCol("X", -0.0f, 0.0f));
+            final Table actual = x.sort("X").head(1);
+            floatToBitsEquals(actual, "X", oneKey(actual), -0.0f);
+        }
+        {
+            final Table x = newTable(floatCol("X", 0.0f, -0.0f));
+            final Table actual = x.sort("X").head(1);
+            floatToBitsEquals(actual, "X", oneKey(actual), 0.0f);
+        }
+    }
+
+    @Test
+    public void testDhc3768_sort_double_results() {
+        {
+            final Table x = newTable(doubleCol("X", -0.0, 0.0));
+            final Table actual = x.sort("X").head(1);
+            doubleToBitsEquals(actual, "X", oneKey(actual), -0.0);
+        }
+        {
+            final Table x = newTable(doubleCol("X", 0.0, -0.0));
+            final Table actual = x.sort("X").head(1);
+            doubleToBitsEquals(actual, "X", oneKey(actual), 0.0);
+        }
+    }
+
+    private static long oneKey(Table table) {
+        final long key;
+        try (final RowSet.Iterator iterator = table.getRowSet().iterator()) {
+            TestCase.assertTrue(iterator.hasNext());
+            key = iterator.nextLong();
+            TestCase.assertFalse(iterator.hasNext());
+        }
+        return key;
+    }
+
+    private static void floatToBitsEquals(Table table, String column, long key, float expectedExact) {
+        final float result = table.getColumnSource(column, float.class).getFloat(key);
+        TestCase.assertEquals(Float.floatToIntBits(expectedExact), Float.floatToIntBits(result));
+    }
+
+    private static void doubleToBitsEquals(Table table, String column, long key, double expectedExact) {
+        final double result = table.getColumnSource(column, double.class).getDouble(key);
+        TestCase.assertEquals(Double.doubleToLongBits(expectedExact), Double.doubleToLongBits(result));
+    }
+
+    private static class CreateAggSpecsDhc3768 implements AggSpec.Visitor {
```

Comment:
This is overkill; hashing is consistent for all aggs. I would consider testing single column, and multi-column.